system/Entity.php

```diff
@@ -359,31 +359,49 @@ public function __set(string $key, $value = null)
 
 		if (! $isNullable || ! is_null($value))
 		{
-			// CSV casts need to be imploded.
-			if ($castTo === 'csv')
+			switch ($castTo)
 			{
-				$value = implode(',', $value);
-			}
-
-			// Array casting requires that we serialize the value
-			// when setting it so that it can easily be stored
-			// back to the database.
-			if ($castTo === 'array')
-			{
-				$value = serialize($value);
-			}
-
-			// JSON casting requires that we JSONize the value
-			// when setting it so that it can easily be stored
-			// back to the database.
-			if (($castTo === 'json' || $castTo === 'json-array') && function_exists('json_encode'))
-			{
-				$value = json_encode($value, JSON_UNESCAPED_UNICODE);
-
-				if (json_last_error() !== JSON_ERROR_NONE)
-				{
-					throw CastException::forInvalidJsonFormatException(json_last_error());
-				}
+				case 'bool':
+				case 'boolean':
+					$value = filter_var($value, FILTER_VALIDATE_BOOLEAN, $isNullable ? FILTER_NULL_ON_FAILURE : 0);
+					break;
+				case 'int':
+				case 'integer':
+					$value = (int) $value;
+					break;
+				case 'float':
+					$value = (float)$value;
+					break;
+				case 'double':
+					$value = (double)$value;
+					break;
```

Comment:
Float and double are the same in PHP
```suggestion
				case 'float':
				case 'double':
					$value = (float) $value;
					break;
```

Reply:
Yes they are. But there is the same logic in castAs method so I wanted to follow it here